logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -54,36 +61,59 @@ def receive(event)
 
   public
   def worker_setup
+    # TODO: Remove this branch, delete this function
     if @workers == 1
+      @is_multi_worker = false
       @worker_plugins = [self]
-      @worker_threads = []
     else
-      define_singleton_method(:handle, method(:handle_worker))
-      @worker_queue = SizedQueue.new(20)
+      @is_multi_worker = true
+      define_singleton_method(:multi_handle, method(:handle_worker))
```

Comment:
Redefining the method seems a bit extreme to me? 
Can we just set the proc to an instance variable and multi_handle proxy the call?